### PR TITLE
feat(tests): expand UI test coverage

### PR DESCRIPTION
**Pull request by damienomurchu**
## Motivation
Expand ui test coverage & add test for landing page. https://issues.jboss.org/browse/AEROGEAR-8849, https://issues.jboss.org/browse/AEROGEAR-8860

## What
Add tests

## Why
Increase test coverage

## How
Add tests

## Verification Steps
Check all tests run in CI for UI

## Checklist:

- [x] Code has been tested locally by PR requester
- [ ] Changes have been successfully verified by another team member 

## Progress

- [x] Finished task
- [ ] TODO

### CODE REVIEW

File: ui/src/components/common/Header.js [hunk 1]
```diff
@@ -26,15 +26,15 @@ export class Header extends React.Component {
   };
 
   onUserDropdownToggle = () => {
-    this.props.toggleHeaderDropdown();
+    this.props.toggleHeaderDropdown(); // TODO - add test coverage
```
Comment (craicoverflow) [inline ui/src/components/common/Header.js]: Should maybe use JSDoc's `@todo` style:

```js
/**
 * @todo Write the documentation.
 * @todo Implement this function.
 */
function foo() {
    // write me
}
```

http://usejsdoc.org/tags-todo.html

Comment (damienomurchu) [reply ui/src/components/common/Header.js]: @craicoverflow good point, if merging in any todo's I will look at adopting this convention. At the moment, just using TODO's to keep track of where I need to add test coverage before merging

File: ui/src/components/__tests__/Header.test.js [hunk 2]
```diff
@@ -1,8 +1,45 @@
...
+describe('Header', () => {
+  const testProps = { currentUser: 'jdoe', isUserDropdownOpen: false, history: [] };
+
+  it('renders the expected components without crashing', () => {
+    const headerComponent = shallow(<Header />);
+    expect(headerComponent).toHaveLength(1);
+  });
+
+  it('has expected props', () => {
+    const wrapper = mount(<Header {...testProps} />);
```
[the remark this answers is not in the document]
Comment (damienomurchu) [reply ui/src/components/__tests__/Header.test.js]: `shallow` will let you render just the node with no children, whereas `mount` will mount the entire component including children nodes, so you can test for sub-components

**Comment by craicoverflow** [comment]
Getting 3 failing tests now:

```
Test Suites: 3 failed, 11 passed, 14 total
Tests:       3 failed, 40 passed, 43 total
Snapshots:   0 total
Time:        7.111s
Ran all test suites matching /a/i.
```

File: ui/src/components/__tests__/Header.test.js [hunk 3]
```diff
@@ -1,8 +1,45 @@
...
+    const props = wrapper.props();
+    console.log('props:', props);
+    expect(props.currentUser).toBeTruthy();
+    expect(props.isUserDropdownOpen).toBeDefined();
```
[the remark this answers is not in the document]
Comment (damienomurchu) [reply ui/src/components/__tests__/Header.test.js]: `isUserDropdownOpen` is a boolean, so can't use truthy.

File: ui/src/components/AppsTable.js [hunk 4]
```diff
@@ -10,4 +10,12 @@ export const AppsTable = ({ columns, rows, sortBy, onSort, onRowClick }) => (
   </Table>
 );
 
+AppsTable.propTypes = {
+  sortBy: PropTypes.object.isRequired,
+  onSort: PropTypes.func.isRequired,
+  columns: PropTypes.array.isRequired,
+  rows: PropTypes.array.isRequired,
+  onRowClick: PropTypes.func.isRequired
```
[the remark this answers is not in the document]
Comment (damienomurchu) [reply ui/src/components/AppsTable.js]: Good point. Let's address this when we refactor, as `AppsTable` may go away

**Comment by damienomurchu** [comment]
@craicoverflow tidied this up, and responded to all suggestion - can you re-review, and will look to merge this in soon

**Pull request merged**